linalg/ode.hpp

```diff
@@ -12,13 +12,93 @@
 #ifndef MFEM_ODE
 #define MFEM_ODE
 
+#include "../general/communication.hpp"
 #include "../config/config.hpp"
 #include "operator.hpp"
-#include "../general/communication.hpp"
+#include <vector>
+#include <memory>
 
 namespace mfem
 {
 
+/// An interface for storing state of previous timesteps
+class ODEStateData
+{
+public:
+   /// Get the maximum number of stored stages
+   virtual int MaxSize() const = 0;
+
+   /// Get the current number of stored stages
+   virtual int Size() const = 0;
+
+   /// Get the ith state vector
+   virtual void Get(int i, Vector &state) const { state = Get(i);};
```

Comment:
You might consider removing this entirely (this is what I believe @v-dobrev was suggesting in an earlier comment)

Reply:
Done.